gramps_webapi/api/resources/person.py

```diff
@@ -1,49 +1,58 @@
 """Person API resource."""
 
-from gramps.gen.display.name import displayer as name_displayer
-
-from .base import (
-    GrampsObjectProtectedResource,
-    GrampsObjectResourceHelper,
-    GrampsObjectsProtectedResource,
-)
-from .util import (
-    get_birthdate,
-    get_birthplace_grampsid,
-    get_citation_grampsids,
-    get_deathdate,
-    get_deathplace_grampsid,
-    get_event_grampsids_roles,
-    get_families_grampsids,
-    get_note_grampsids,
-    get_parents_grampsids,
-)
+import gramps.gen.filters.rules.person as rule_classes
+
+from .base import (GrampsObjectProtectedResource, GrampsObjectResourceHelper,
+                   GrampsObjectsProtectedResource)
```

Comment:
This is not formatted with `black`; are you using the pre-commit hook?

Reply:
In case you are using `isort`, it is always best to first sort and then format, since `isort` has its own ideas about formatting that are different from `black`.

Reply:
Still not setup right, need to look closer at it.

Reply:
(gramps_webapi) [webapi@abyss web-api]$ more .pre-commit-config.yaml
repos:
  - repo: https://github.com/psf/black
    rev: stable
    hooks:
      - id: black
  - repo: https://github.com/pre-commit/mirrors-isort
    rev: v4.3.21
    hooks:
      - id: isort
  - repo: https://github.com/PyCQA/pylint
    rev: pylint-2.5.3
    hooks:
      - id: pylint
        stages: [commit]
        args:
          - --score=n
  - repo: https://github.com/PyCQA/pydocstyle
    rev: 5.0.2
    hooks:
      - id: pydocstyle
        stages: [commit]

I will reverse order.

Reply:
Oh indeed, looks like I screwed that up, sorry :hand_over_mouth:

Reply:
Actually, after running pre-commit install to install the hook, which we need to update README for maybe, I can't commit because they all seem to conflict with each other.  So I disabled that and ran manually again like I was. Maybe I'm missing something.

Reply:
isort fails because it reformatted what black did, and black fails because it reformatted what isort did. 

pylint is unhappy about lots of stuff.  Do we want to fix as much of that as possible?  I can look at that more later tonight after church.

Reply:
Although some of what pylint seems to not like is what black and isort did. I imagine there must be a way to skip some checks to make them all play nice.

Reply:
We can modify the `.pylintrc` file.  What doesn't it like?

Reply:
Numerous wrong hanging indentations before blocks are the result of black reformatting.
db as both an attribute and variable doesn't conform to snake_case.
Some classes have too few public methods since they currently have none.
Too many return statements.
Too many branches.
Some duplicate code across files.
Use of [] and {} is dangerous as default arguments.
Method object_extend is abstract in GrampsObjectResourceHelper but not overridden, a few unused imports.

Reply:
Added a .pylintrc to disable C0330 and C0326 as per recommendations here: https://black.readthedocs.io/en/stable/compatible_configs.html

Also updated version of isort being used and set it to use the black compatible profile. 

Added mypy to the pre-commit config and cleaned some stuff up it was not happy about. 

Added a .flake8 but did not add it to the pre-commit config. I think maybe we should.  I did run it manually though and fixed an issue it found that pylint did not. It is also complaining about a naked except: in gramps_webapi/auth.py but left that alone for now.

Was able to get the pylint exit code down from 30 to 10, but not there yet.

Reply:
Oh my oh my. I was curious and just started running pylint against the Gramps codebase...

Reply:
Yeah that's why I think enforcing formatting and being super strict is helpful when starting a project from scratch :slightly_smiling_face:

Reply:
I am really liking this pre-commit setup, especially black.

Reply:
Some of the pylint complaints I'm a little unsure how to go about resolving.  Do you them all resolved before a merge?  And what else would you like to see cleaned up before one?

Reply:
Also, going to start on apispec.yaml updates today.

Reply:
> Added a .pylintrc to disable C0330 and C0326

It is better to use readable settings rather than codes.  I have created PR #27 which adds the settings to a default configuration file.

Reply:
> Added a .flake8 but did not add it to the pre-commit config. I think maybe we should.

The main Gramps project uses pylint rather than flake8.

Reply:
> Do you want them all resolved before a merge?

No.  Use pylint for guidance.  We aim for a pylint score of 9.0 or higher in new code.

However, if there are good reasons to write code with warnings then do so.  There is no need to disable a warning locally in the code.  I would rather be able to see it.

A good example is avoiding the use of variables `x`, `y` and `z`.  These are normally not good choices for variable names, but if you are writing graphics code then they may be appropriate.